CondFormats/HGCalObjects/interface/HGCalESProducerTools.h

```diff
@@ -0,0 +1,97 @@
+#ifndef CondFormats_HGCalObjects_HGCalESProducerTools_h
+#define CondFormats_HGCalObjects_HGCalESProducerTools_h
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <string>
+#include <sstream>    // for std::istringstream
+#include <fnmatch.h>  // to match glob patterns
+#include <nlohmann/json.hpp>
```

Comment:
This necessitates
```xml
<use name="json"/>
```
to `CondFormats/HGCalObjects/BuildFile.xml`

Reply:
Actually, what motivates placing these function in `CondFormats/HGCalObjects`? AFAICT they are presently used only in `RecoLocalCalo/HGCalRecAlgos`. The functions seem to be intimately tied to the JSON presentation of the data. Is JSON planned to be used for the eventual CondDB storage?

Reply:
It is a bit early to say if whether the final format will be JSONs, but I did intentionally put the help functions to match module typecodes to keys in `CondFormats/HGCalObjects` instead `RecoLocalCalo/HGCalRecAlgos` because I thought we might keep this way of matching keys (the typecode contains information about the sensor thickness, number of channels, etc.).

Reply:
From dependency management point of view it would be better to keep the dependencies in the smallest possible scope. Presently we have 13 packages that depend (directly or indirectly) on `json`, this change would add 28 packages more there.

The file naming of "tools to produce the condformats" also hints towards the "condformats" not to be the best placement (I mean, we separate the production of data from the data formats on purpose).

Reply:
Okay. I'll move `CondFormats/HGCalObjects/interface/HGCalESProducerTools.h` to `RecoLocalCalo/HGCalRecAlgos/interface/HGCalESProducerTools.h`.

Reply:
Done in https://github.com/cms-sw/cmssw/pull/48137/commits/5ba71768546ae894cbffa898b9c8b1bd9cad9a5b.